src/test/ui/feature-gates/feature-gate-generic_arg_infer.rs

```diff
@@ -0,0 +1,10 @@
+// run-pass
```

Comment:
this test doesn't actually test whether `generic_arg_infer` is needed, does it?

Don't we want this test to **not** use this feature and fail because of it?

Reply:
added revisions to show that it works with it, and fails without it